include/blas_3d.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <quda_internal.h>
+#include <color_spinor_field.h>
+
+namespace quda
+{
+
+  namespace blas3d
+  {
+
+    // Local enum for the 3D copy type
+    enum copy3dType { COPY_TO_3D, COPY_FROM_3D, SWAP_3D };
```

Comment:
I've been trying to use `enum class` for internal (i.e., non-`quda_enum.h`) enums, I think we should apply that here.

Reply:
[6383489](https://github.com/lattice/quda/pull/1506/commits/63834893048dbee7acee91f788bbc14003d933cc)